Alarms.cpp

```diff
@@ -191,16 +206,66 @@ int AlarmManager::numON() const {
   return num;
 }
 
-String AlarmManager::getText() const {
-  const int num_on = numON();
+int AlarmManager::numON_Confirm() const {
+  int num = 0;
+  bool aConfirmAlarm;
+  for (int i = 0; i < NUM_ALARMS; i++) {
+    aConfirmAlarm = (i==NOT_CONFIRM_TV || i==NOT_CONFIRM_RR ||
+                     i==NOT_CONFIRM_IE || i==NOT_CONFIRM_AC);
+    if(aConfirmAlarm) {
+      num += (int)alarms_[i].isON();
+    }
+  }
+  return num;
+}
+
+int AlarmManager::numON_NonConfirm() const {
+  int num = 0;
+  bool aConfirmAlarm;
+  for (int i = 0; i < NUM_ALARMS; i++) {
+    aConfirmAlarm = (i==NOT_CONFIRM_TV || i==NOT_CONFIRM_RR ||
+                     i==NOT_CONFIRM_IE || i==NOT_CONFIRM_AC);
+    if(!aConfirmAlarm) {
+      num += (int)alarms_[i].isON();
+    }
+  }
+  return num;
+}
+
+String AlarmManager::getGeneralAlarmText() const {
+  const int num_on = numON_NonConfirm();
+  String text = "";
+  if (num_on > 0) {
+    // determine which of the on alarms to display
+    const int index = millis() % (num_on * kDisplayTime) / kDisplayTime;
+    int count_on = 0;
+    int i;
+    bool aConfirmAlarm;
+    for (i = 0; i < NUM_ALARMS; i++) {
+      aConfirmAlarm = (i==NOT_CONFIRM_TV || i==NOT_CONFIRM_RR ||
+                       i==NOT_CONFIRM_IE || i==NOT_CONFIRM_AC);
+      if (!aConfirmAlarm && alarms_[i].isON()) {
+        if (count_on++ == index) break;
+      }
+    }
+    text = alarms_[i].text();
+  }
+  return text;
+}
+
+String AlarmManager::getUnconfirmedKnobText() const {
+  const int num_on = numON_Confirm();
   String text = "";
   if (num_on > 0) {
     // determine which of the on alarms to display
     const int index = millis() % (num_on * kDisplayTime) / kDisplayTime;
     int count_on = 0;
     int i;
+    bool aConfirmAlarm;
     for (i = 0; i < NUM_ALARMS; i++) {
-      if (alarms_[i].isON()) {
+      aConfirmAlarm = (i==NOT_CONFIRM_TV || i==NOT_CONFIRM_RR ||
+                       i==NOT_CONFIRM_IE || i==NOT_CONFIRM_AC);
+      if (aConfirmAlarm && alarms_[i].isON()) {
```

Comment:
This is confusing and brittle. Now that the confirm knobs and alarms are being written to different places, they should be treated differently. But hardcoding the list of which alarms are confirm alarms, and writing these almost duplicate functions is bad practice. We should abstract the logic of which alarms are in which cycle, so we don't need to hardcode checks and write code multiple times